draft-irtf-nwcrg-coding-and-congestion-01.txt

```diff
@@ -1,504 +1,528 @@
+
+6.  Server-side reaction to repaired packet signals
+
+   [MICHAEL: I would remove this section entirely: it only adds to
+   confusion.  If the inputs are seen as distinct, and we say that the
+   raw inputs shouldn't be shared, then why would one even think about
+   what delay-based CC does with this input?  It's just not the right
+   input.] Delay-based congestion controls ignore packets that have been
+   repaired with coding.  There is no need to define best current
+   pratices in this case.  However, more discussions are required for
+   congestion controls that use loss as congestion signals (potentially
+   among other congestion detection mechanism).
+
+6.1.  The server congestion control considers repaired packet signals as
+      congestion-implied packet losses
+
+   [MICHAEL: As above, lets remove all of this please.  This text gets
+   in the way of a clean separation.] In this solution, the server
+   reacts to repaired packet signals as to congestion-implied packet
+   losses.  That being said, this does not necessarily mean that the
+   packets have actually been lost.  The server may have other means to
+   identify that the packet was just out-of-order and ignore the
+   repaired packet signals.
+
+   The advantages of the solution are (1) that coding mechanisms do not
+   hide congestion signals, such as packets voluntary dropped by a AQM
+   [RFC7567] and (2) packets may be repaired faster than with a
+   traditional retransmission mechanisms.
+
+   The drawback of this solution is that, if there is a high non-
+   congestion loss rate, the congestion control throughput may decrease
+   drastically.  Reporting this amount of loss to a server may reduce
+   the application goodput when there is no actual congestion.
+
+6.2.  The server adapts its window reduction to repaired packet signals
+
+   [MICHAEL: this is even just a terrible idea.  There's no good
+   rationale to back off less when a packet is repaired.] In this
+   solution, the server does not reduce the congestion window with the
+   same amount when the "repaired packet" signal is received, i.e.  when
+   a packet has been lost but repaired.  Example of this solution could
+   be based on [RFC8511] or considering that recovering an isolated
+   packet is not an actual sign of congestion.
+
+
+
+
+Kuhn, Lochin, Michel & WeExpires July 11, 2020                  [Page 6]
+
+Internet-Draft           Coding and congestion              January 2020
+
+
+   The advantage of the solution is that in cases where there is no
+   actual congestion, coding could help in improving the transmission
+   without ignoring congestion signals.
+
+   The main drawback is the precised design of the solution and its
+   interaction with AQM mechanisms [RFC7567].  Moreover there may be
+   fairness issues since AIMD convergence may not be guaranteed.
+
+6.3.  The server ignores repaired packet signals
+
+   [MICHAEL: as above, also here, the signals are discussed as going

```

Comment:
I agree, I think however that such a behaviour is referring to solutions I cited in my comment above (using a coded tunnel to speed-up TCP connections  where losses should not be due to congestion).